pkg/stats/flush_notifier.go

```diff
@@ -35,7 +36,7 @@ func (fn *flushNotifier) RegisterFlush() (ch <-chan time.Duration, unregister fu
 
 // NotifyFlush will notify any registered channels that a flush has completed.
 // Non-blocking, thread-safe.
-func (fn *flushNotifier) NotifyFlush(d time.Duration) {
+func (fn *flushNotifier) NotifyFlush(ctx context.Context, d time.Duration) {
```

Comment:
`ctx` isn't used within this function

Reply:
It was added to the Statser interface so that InternalStatser had a context, which it needs to drain the MetricConsolidator.

Reply:
Could you not add it to the `select` block done bellow?

Reply:
Ahh, I see what you mean.  Technically yes, but there's really no point. The loop is basically non-blocking because of the `default:` clause.

Remember the primary purpose of cancelling a context is to signal a long-running call to finish - this isn't long running.